src/software/world/team.cpp

```diff
@@ -178,6 +178,31 @@ std::vector<Robot> Team::getAllRobotsExceptGoalie() const
     return all_robots;
 }
 
+std::optional<Robot> Team::getNearestRobot(const Point& ref_point) const
+{
+    return getNearestRobot(this->getAllRobots(), ref_point);
+}
+
+std::optional<Robot> Team::getNearestRobot(const std::vector<Robot>& robots,
+                                           const Point& ref_point) const
```

Comment:
can we combine these functions to simplify this, i.e. put the contents of `getNearestRobot(const std::vector<Robot>& robots, const Point& ref_point)` into `getNearestRobot(const Point& ref_point)`?

Reply:
can you explain a little bit more what you mean by this?

Reply:
@jared-paul I think what Jon means is that all the top method does is that it calls the bottom one, so they can be combined into one.
@jonathanlew I've combined them. Though I thought we might want to keep them separate in case we want to call `getNearestRobot(const std::vector<Robot>& robots, const Point& ref_point)` with a different set of `robots`. Though I'm not sure if that's ever gonna be the case.

Reply:
@jonathanlew Looking further into it, it seems like both methods are being used. (eg. in [`ai/evaluation/enemy_threat.cpp` ](https://github.com/UBC-Thunderbots/Software/pull/1822/files#diff-bb75335fcfd1c0d8b3b8fc790403e12edf8494c4f84a3f5b8a7b2c2c15e9c641R130-R131) `getNearestRobot(const std::vector<Robot>& robots, const Point& ref_point)` is being used)

Reply:
OK, can we make this function `static`?